packages/react-components/src/components/chart/store/highlightedDataStreams.ts

```diff
@@ -0,0 +1,25 @@
+import { DataStream } from '@iot-app-kit/core';
+import { StateCreator } from 'zustand';
+
+export interface HighlightedDataStreamsData {
+  highlightedDataStreams: DataStream[];
+}
+
+export interface HighlightedDataSteamsState extends HighlightedDataStreamsData {
+  highlightDataStream: (datastream?: DataStream) => void;
+  unHighlightDataStream: (datastream?: DataStream) => void;
+}
+
+export const createHighlightedDataStreamsSlice: StateCreator<HighlightedDataSteamsState> = (set) => ({
+  highlightedDataStreams: [],
+  highlightDataStream: (datastream?: DataStream) =>
+    set((state) => {
+      if (!datastream) return { highlightedDataStreams: state.highlightedDataStreams };
```

Comment:
this should be if ```(!datastream) return state ;``` right ?

Reply:
Yes, this is the same thing. But you're right, probably better to change it to just be state